Makefile

```diff
@@ -17,11 +17,27 @@ else
 GOBIN=$(shell go env GOBIN)
 endif
 
+# Setting SHELL to bash allows bash commands to be executed by recipes.
+# This is a requirement for 'setup-envtest.sh' in the test target.
+# Options are set to exit when a recipe line exits non-zero or a piped command fails.
+SHELL = /usr/bin/env bash -o pipefail
+.SHELLFLAGS = -ec
+
 all: manager
 
 # Run tests
-test: fmt vet # generate manifests
-	go test ./... -coverprofile coverage.out
+# This is the path to save binary files used in integration test.
+ENVTEST_ASSETS_DIR=$(shell pwd)/testbin
+
+# Run all unit tests.
+test: manifests generate fmt vet
+	 go test --tags=unit ./... -coverprofile coverage.out
+
+# Run tekton backend controller integration test
+tekton-support-integration-test:
+	mkdir -p ${ENVTEST_ASSETS_DIR}
+	test -f ${ENVTEST_ASSETS_DIR}/setup-envtest.sh || curl -sSLo ${ENVTEST_ASSETS_DIR}/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.6.3/hack/setup-envtest.sh
+	source ${ENVTEST_ASSETS_DIR}/setup-envtest.sh; fetch_envtest_tools $(ENVTEST_ASSETS_DIR); setup_envtest_env $(ENVTEST_ASSETS_DIR); export ACK_GINKGO_DEPRECATIONS=1.16.2; go test --tags=integration ./controllers/tekton
```

Comment:
No longer need to set environment variable `TEST_USE_EXISTING_CLUSTER` as `true` in the integration test.

Reply:
`--tags` keyword was added to run the test to differentiate unit test or integration test.
If `tags=integration`, we will run integration test.
If `tags=unit`, we will run integration test.
This is controlled by adding `// + build tag` at the head of the test files.

Reply:
Can you provide a document or comment to describe how to tag as integration test?